xlators/protocol/server/src/server.c

```diff
@@ -871,7 +871,9 @@ server_reconfigure(xlator_t *this, dict_t *options)
     this->ctx->statedump_path = gf_strdup(statedump_path);
 
 do_auth:
-    if (!conf->auth_modules)
+    if (conf->auth_modules)
+        gf_auth_fini(conf->auth_modules);
+    else
         conf->auth_modules = dict_new();
```

Comment:
Should this be instead something like below?

```suggestion
    if (conf->auth_modules) {
        gf_auth_fini(conf->auth_modules);
    }
    conf->auth_modules = dict_new();
```

Reply:
IIUC no, and your approach causes a dict leak (which may be fixed by adding `dict_unref(conf->auth_modules)` after `gf_auth_fini(conf->auth_modules)`). If we do not recreate `conf->auth_modules`, it will be just repopulated with new key/value pairs in `get_auth_types()`.